src/rules/objectLiteralShorthandRule.ts

```diff
@@ -63,6 +98,32 @@ function walk(ctx: Lint.WalkContext<void>) {
     });
 }
 
+function fixShorthandToLonghand(node: ts.ShorthandPropertyAssignment | ts.MethodDeclaration): Lint.Fix {
+    const isGenerator = (node as ts.MethodDeclaration).asteriskToken !== undefined;
+    const isAsync = hasModifier(node.modifiers, ts.SyntaxKind.AsyncKeyword);
+    let
+    replacementStart = (isAsync && node.modifiers !== undefined) ? getModifier(node, ts.SyntaxKind.AsyncKeyword)!.getStart() : -1;
+    replacementStart = (isGenerator && !isAsync) ? (node as ts.MethodDeclaration).asteriskToken!.getStart() : -1;
+    replacementStart = replacementStart === -1 ? node.name.getStart() : replacementStart;
```

Comment:
the `async` keyword is the only modifier allowed here, so `replacementStart` can always be `node.getStart()`